labs/test/tsc/defaults.bzl

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A docstring"""
+
+load("//labs:tsc.bzl", _tsc = "tsc")
+
+def tsc(srcs = [], declarations = True, sourcemaps = True, **kwargs):
+    """A docstring"""
+
+    # Define optional pre-declared output labels
+    outs = []
+    for src in srcs:
```

Comment:
This is for pre-declared outputs which is an optional feature for this rule. It lets you use labels to select individual output files from the target. There is no way that I could find to get pre-declared outputs from the rule itself but its not strictly needed.

Reply:
The rule will work just find without specifying any `outs` but there will be no pre-declared output labels in that case.

Reply:
It also lets you specify additional outputs that the rule may not be away of if you are using a tool such as `ngc` instead or your tsconfig results in other outs besides js, d.ts & map files.

Reply:
Ah right! Yeah that makes sense.
I think leaving it optional is fine for people who want to extend it, since most use cases wont need the labels for outs.